Realm/RLMUtil.mm

```diff
@@ -436,7 +389,64 @@ BOOL RLMIsRunningInPlayground() {
     return [[NSBundle mainBundle].bundleIdentifier hasPrefix:@"com.apple.dt.playground."];
 }
 
-id RLMMixedToObjc(realm::Mixed const& mixed) {
+realm::Mixed RLMObjcToMixed(__unsafe_unretained id v, __unsafe_unretained RLMRealm *realm) {
+    if (!v || v == NSNull.null) {
+        return realm::Mixed();
+    }
+
+    RLMPropertyType type;
+    if ([v isKindOfClass:[RLMValueBase class]]) {
+        v = [v rlmValue];
+        type = [v valueType];
+    }
+    else if ([v conformsToProtocol:@protocol(RLMValue)]) {
+        type = [v valueType];
+    }
+    else {
+        REALM_TERMINATE("Unexpected Type");
+    }
+
+    switch (type) {
+        case RLMPropertyTypeInt:
+            return realm::Mixed([(NSNumber *)v intValue]);
```

Comment:
All the uses of `intValue` need to be `longLongValue` to avoid truncating 64-bit ints.